plugins/prometheus/pkg/data/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 FairwindsOps Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package data
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+)
+
+// Owner is the information about a pod that a set of metrics belongs to.
+type Owner struct {
+	Container           string
+	PodName             string
+	ControllerNamespace string
+	ControllerName      string
+	ControllerKind      string
+}
+
+// Statistics is an aggregation of the metrics by pod/container
+type Statistics struct {
+	Owner
+	StartTime  time.Time
+	Metric     string
+	Value      int64
+	Request    int64
+	LimitValue int64
+}
+
+// CombinedRequest is the cpu/memory and requests for a given pod/container
+type CombinedRequest struct {
+	Owner
+	cpu           []model.SamplePair
+	memory        []model.SamplePair
+	memoryRequest model.SampleValue
+	cpuRequest    model.SampleValue
+	memoryLimit   model.SampleValue
+	cpuLimit      model.SampleValue
```

Comment:
is it possible to observe changes in limits/requests over time? That'd be a huge increase in data storage, but could make things more accurate...

Reply:
Yep, we're going to be sending up requests/limits for each so we'll be able to visualize that.

Reply:
So why are these a single value, versus slices above?

Reply:
Oh because an individual pod's request/limits shouldn't ever change. So we'll track the request/limits changes of the controller over time by capturing the request/limits of these individual pods. There might be some edge cases with static pods or naked pods where a new pod with the same name is created. But that should only really mess with data like right at the margins of the 10 minute windows we're running this job on. https://github.com/FairwindsOps/insights-plugins/blob/fe07030c930d7249daecce86651e9450088b875d/plugins/prometheus/pkg/data/types.go#L32 is the actual type getting saved off. CombinedRequest is just a helper class to merge the data to make it easier to process (probably not even that useful anymore)

Reply:
Ohh that makes so much more sense. Thanks!